### PR TITLE
Fix access rights on employee report

### LINKED ISSUE #462: Salary report account rights
- [x] If any period in the selected year is as low as (2)Open/Åpen, every person with access level 2 or above  should see add, edit, lock and delete links
- [x] If any period in the selected year is as low as (3)Done/Ferdig, every person with access level 3 or above  should see add, edit, lock and delete links
- [x] If all periods in the selected year are (4)Closed/Stengt, no links should be shown
- [x] Every person with access level 4 should see the unlock link all the time except when all periods of the year are (4)Closed/Lukket

### PR DESCRIPTION
**Pull request by mladjo2505**
Closes #462

### CODE REVIEW

File: modules/salary/view/employeereport.php [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <?php
 $year = isset($_REQUEST["year"]) ? (int)$_REQUEST["year"] : (isset($_REQUEST["report_year"]) ? (int)$_REQUEST["report_year"] : date("Y"));
+$max_period_status = $_lib['db']->get_row(array('query' => "SELECT MAX(Status) AS Max FROM accountperiod WHERE Period LIKE '$year-%'"));
+$required_access_level = $max_period_status->Max;
+if ($required_access_level == 4) {
```
Comment (mladjo2505) [inline modules/salary/view/employeereport.php]: Add comment

File: modules/salary/view/employeereport.php [hunk 2]
```diff
@@ -301,35 +309,32 @@ function print_sums($employee_id, $codes, $report) {
...
+            <td><a href="%slock_account_report&t=salary.employeereport&SalaryReportAccountID=%d&year=%d" onclick=\'return confirm("Er du sikker p&aring; at du vil l&aring;se?")\'>l&aring;s</a></td>
+            <td><a href="%sdelete_account_report&t=salary.employeereport&SalaryReportAccountID=%d&year=%d"  onclick=\'return confirm("Er du sikker p&aring; at du vil slette?")\'>slett</a></td>',
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year,
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year,
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year
         );
-
     }
-    else {
+    if ($accountreport['Locked'] == 1) {
```
Comment (mladjo2505) [inline modules/salary/view/employeereport.php]: elseif

File: modules/salary/view/employeereport.php [hunk 3]
```diff
@@ -446,31 +452,29 @@ function print_sums($employee_id, $codes, $report) {
...
+            <td><a href="%slock_account_report&t=salary.employeereport&SalaryReportAccountID=%d&year=%d" onclick=\'return confirm("Er du sikker p&aring; at du vil l&aring;se?")\'>l&aring;s</a></td>
+            <td><a href="%sdelete_account_report&t=salary.employeereport&SalaryReportAccountID=%d&year=%d"  onclick=\'return confirm("Er du sikker p&aring; at du vil slette?")\'>slett</a></td>',
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year,
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year,
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year
         );
 
     }
-    else {
+    if ($accountreport['Locked'] == 1) {
```
Comment (mladjo2505) [inline modules/salary/view/employeereport.php]: elseif

File: modules/salary/view/employeereport.php [hunk 4]
```diff
@@ -17,33 +22,32 @@ function print_values($codes, $line, $print_extra = true) {
 
     printf("<td>%s</td>", $line['Comment']);
 
-    if($print_extra) {
+    if($print_extra && $_lib['sess']->get_person('AccessLevel') >= $required_access_level) {
```
Comment (mladjo2505) [inline modules/salary/view/employeereport.php]: restructure if

File: modules/salary/view/employeereport.php [hunk 5]
```diff
@@ -520,31 +524,29 @@ function print_sums($employee_id, $codes, $report) {
...
+            <td><a href="%slock_account_report&t=salary.employeereport&SalaryReportAccountID=%d&year=%d" onclick=\'return confirm("Er du sikker p&aring; at du vil l&aring;se?")\'>l&aring;s</a></td>
+            <td><a href="%sdelete_account_report&t=salary.employeereport&SalaryReportAccountID=%d&year=%d"  onclick=\'return confirm("Er du sikker p&aring; at du vil slette?")\'>slett</a></td>',
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year,
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year,
             $_lib['sess']->dispatch, $accountreport['SalaryReportAccountID'], $year
         );
 
     }
-    else {
+    if ($accountreport['Locked'] == 1) {
```
Comment (mladjo2505) [inline modules/salary/view/employeereport.php]: elseif

**Review by sls-lnluksa — APPROVED** [review]
⚠️  @stoivo @agvald 
Issue is a bit unclear.
About unlock button, should that be always displayed for user with access level 4, or display only if user is access level 4 and there are no locked periods?

**Comment by agvald** [comment]
Unlock button should be seen by users with access level 4 when any period of the year is different than closed/lukket 4

**Review by agvald — APPROVED** [review]

**Review by stoivo — APPROVED** [review]

**Pull request merged**